### DISCUSSION TITLE
Help with Setting Up Personal Repo

### DISCUSSION THREAD

**Discussion by hesam-init** [void-linux/void-packages#50888] [Q&A]
Hey everyone, 

I tried to create a personal repository for the packages I need. I set up a GitHub Action to build the packages and then push them to a branch to use as a remote repository. Here is my repo:

[https://github.com/void-land/void-repository](https://github.com/void-land/void-repository)

However, when I attempt to install a package from the repo, I get this RSA error:

![RSA Error](https://github.com/void-linux/void-packages/assets/89181937/a868a567-0a0b-4252-a5d3-85d0ac512c45)

Can anyone help with this issue?

**Comment by Chocimier** [comment]
https://github.com/void-linux/void-packages#sharing-and-signing

**Comment by Chocimier** [comment]
You are generating key every run of workflow, what may be problematic. It would be easier if key was fixed for the repo.

By upload date, repodata, signature and package seem to use same key, but still check:
Did you accept that key as signing the repo? Did you removed package and signature from local cache to make sure all three match after download?

**Comment by ndpm13** [comment]
I have a repo that was working fine before, but now there is an update that doesn't allow the root user to build packages and that broke everything, if you found a good workflow please let us know T^T

![image](https://github.com/user-attachments/assets/5cd67a10-9d35-48ce-959f-f53997d92659)

**Reply by classabbyamp** [reply]
root was never allowed to build packages unless using the ethereal chroot style. nothing has changed with that

**Reply by ndpm13** [reply]
...fr? So all I have to do is set an env var to use the ethereal thingy?
